subsys/bluetooth/host/att.c

```diff
@@ -71,6 +71,26 @@ NET_BUF_POOL_DEFINE(prep_pool, CONFIG_BT_ATT_PREPARE_COUNT, BT_ATT_MTU,
 		    sizeof(struct bt_attr_data), NULL);
 #endif /* CONFIG_BT_ATT_PREPARE_COUNT */
 
+struct gatt_params {
+	union {
+		struct bt_gatt_notify_params notify;
+		struct bt_gatt_indicate_params indicate;
+#if defined(CONFIG_BT_GATT_CLIENT)
+		struct bt_gatt_exchange_params exchange;
+		struct bt_gatt_discover_params discover;
+		struct bt_gatt_read_params read;
+		struct bt_gatt_write_params write;
+		struct bt_gatt_subscribe_params subscribe;
+#endif /* CONFIG_BT_GATT_CLIENT */
+	};
+};
+
+#define ATT_REQ_MAX_PARAMS \
+	(sizeof(struct bt_att_req) + sizeof(struct gatt_params))
+
+K_MEM_POOL_DEFINE(req_pool, sizeof(struct bt_att_req), ATT_REQ_MAX_PARAMS,
```

Comment:
I am now wondering if this should be optional, since applications might be reusing the ATT request memory elsewhere and now we are dedicating it solely to this. It would complicate a bit the return values (0 or id, maybe we could make `id == 0` an invalid ID?)

Reply:
From the feedback I got the maintaining of the requests on the application can get a little out of hand depending on the how many they are, and it is not the simple to detect if a request is in use, this actually makes testing a lot simpler since one can just make as many request as it wants without worrying if the parameters are not being overwritten or something. If I had to rewrite the test to have every single parameter stored in a static the tests would be somewhat different, or I would have to perform them twice one with parameters copy enable and another without which I don't thing is a complexity we'd like to have.